src/git.rs

```diff
@@ -86,8 +97,19 @@ pub(crate) fn create(project_dir: &Path, args: GitConfig) -> Result<String> {
             //  - https://github.com/rust-lang/cargo/issues/8468
             let repo = git2::Repository::init(&temp.path())?;
             let mut git_remote = repo.remote_anonymous(remote.url().as_str())?;
-            git_remote.connect(git2::Direction::Fetch)?;
-            let default_branch = git_remote.default_branch()?;
+
+            let default_branch = if is_http {
+                git_remote.connect(git2::Direction::Fetch)?;
+                git_remote.default_branch()?
+            } else {
+                let mut cb = RemoteCallbacks::new();
```

Comment:
question: is that not a bit of code duplication to like 135?

Reply:
`git2::RemoteCallbacks` doesn't implement `Clone` or `Copy` moved the creation to a function that returns a RemoteCallback instead.